haystack/nodes/retriever/_embedding_encoder.py

```diff
@@ -125,6 +157,22 @@ def embed_documents(self, docs: List[Document]) -> List[np.ndarray]:
         passages = [[d.meta["name"] if d.meta and "name" in d.meta else "", d.content] for d in docs]  # type: ignore
         return self.embed(passages)
 
+    def train(self, training_data: List[Dict[str, Any]], batch_size: int = 16) -> None:
+        train_examples = [
+            InputExample(texts=[i["question"], i["pos_doc"], i["neg_doc"]], label=i["score"]) for i in training_data
+        ]
+        logger.info(f"GPL training/adapting {self.embedding_model} with {len(train_examples)} examples")
+        train_dataloader = DataLoader(train_examples, batch_size=batch_size, drop_last=True, shuffle=True)
+        train_loss = losses.MarginMSELoss(self.embedding_model)
+
+        # Tune the model
+        self.embedding_model.fit(
+            train_objectives=[(train_dataloader, train_loss)], epochs=1, warmup_steps=int(len(train_dataloader) * 0.1)
```

Comment:
I think this shouldn't be limited to training for one epoch. They say they train for 100k steps in the paper. You can do that in a single epoch if you have enough data, but I'm not sure this should be the only option.